sdks/python/apache_beam/examples/leader_board_it_test.py

```diff
@@ -0,0 +1,160 @@
+    self.input_sub = self.input_topic.subscription(unique_subscrition_name)
+
+    self.input_topic.create()
+    test_utils.wait_for_topics_created([self.input_topic])
+    self.input_sub.create()
+
+    # Set up BigQuery environment
+    from google.cloud import bigquery
+    client = bigquery.Client()
+    unique_dataset_name = self.OUTPUT_DATASET + str(int(time.time()))
+    self.dataset = client.dataset(unique_dataset_name, project=self.project)
+    self.dataset.create()
+
+    self._test_timestamp = int(time.time() * 1000)
+
+  def _inject_pubsub_game_events(self, topic, message_count):
+    """Inject game events as test data to PubSub."""
+
+    logging.debug('Injecting %d game events to topic %s',
+                  message_count, topic.full_name)
+
+    for _ in xrange(message_count):
+      topic.publish(self.INPUT_EVENT % self._test_timestamp)
+
+  def _cleanup_pubsub(self):
+    test_utils.cleanup_subscriptions([self.input_sub])
+    test_utils.cleanup_topics([self.input_topic])
+
+  def tearDown(self):
+    self._cleanup_pubsub()
+
+  @attr('IT')
+  def test_leader_board_it(self):
+    state_verifier = PipelineStateMatcher(PipelineState.RUNNING)
+
+    success_condition = 'total_score=5000 LIMIT 1'
+    users_query = ('SELECT total_score FROM [%s:%s.%s] '
+                   'WHERE %s' % (self.project,
+                                 self.dataset.name,
+                                 self.OUTPUT_TABLE_USERS,
+                                 success_condition))
+    bq_users_verifier = BigqueryMatcher(self.project,
+                                        users_query,
+                                        self.DEFAULT_EXPECTED_CHECKSUM)
+
+    teams_query = ('SELECT total_score FROM [%s:%s.%s] '
+                   'WHERE %s' % (self.project,
+                                 self.dataset.name,
+                                 self.OUTPUT_TABLE_TEAMS,
+                                 success_condition))
+    bq_teams_verifier = BigqueryMatcher(self.project,
+                                        teams_query,
+                                        self.DEFAULT_EXPECTED_CHECKSUM)
+
+    extra_opts = {'subscription': self.input_sub.full_name,
+                  'dataset': self.dataset.name,
+                  'topic': self.input_topic.full_name,
+                  'team_window_duration': 1,
+                  'wait_until_finish_duration':
+                      self.WAIT_UNTIL_FINISH_DURATION,
+                  'on_success_matcher': all_of(state_verifier,
+                                               bq_users_verifier,
+                                               bq_teams_verifier)}
+
+    # Register cleanup before pipeline execution.
+    self.addCleanup(utils.delete_bq_table, self.project,
+                    self.dataset.name, self.OUTPUT_TABLE_USERS)
+    self.addCleanup(utils.delete_bq_table, self.project,
+                    self.dataset.name, self.OUTPUT_TABLE_TEAMS)
+
```

Comment:
Do you want to move `_cleanup_pubsub` here as well? That way (a) there will be a single place for all clean up code (b) you can explicitly know the clean up execution order.